### PR TITLE
[MM-16782] Migrate "Team.GetTeamMembersForExport" to Sync by default

### PR DESCRIPTION
**Pull request by Hyaxia**
Summary
Migrate `Team.GetTeamMembersForExport` to sync by default.

Fixes https://github.com/mattermost/mattermost-server/issues/11546

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.PHONY: build package run stop run-client run-server stop-client stop-server restart restart-server restart-client start-docker clean-dist clean nuke check-style check-client-style check-server-style check-unit-tests test dist setup-mac prepare-enteprise run-client-tests setup-run-client-tests cleanup-run-client-tests test-client build-linux build-osx build-windows internal-test-web-client vet run-server-for-web-client-tests
+PHONY: build package run stop run-client run-server stop-client stop-server restart restart-server restart-client start-docker clean-dist clean nuke check-style check-client-style check-server-style check-unit-tests test dist setup-mac prepare-enteprise run-client-tests setup-run-client-tests cleanup-run-client-tests test-client build-linux build-osx build-windows internal-test-web-client vet run-server-for-web-client-tests
```
Comment (jespino) [inline Makefile]: Can you revert this change?

Comment (Hyaxia) [reply Makefile]: Yeah sorry, dont know how it got there 😅
